ironfish/src/merkletree/merkletree.ts

```diff
@@ -453,8 +455,22 @@ export class MerkleTree<
     return this.db.withTransaction(tx, async (tx) => {
       const leafCount = await this.getCount('Leaves', tx)
 
-      if (leafCount === 0 || pastSize > leafCount || pastSize === 0) {
-        throw new Error(`Unable to get past size ${pastSize} for tree with ${leafCount} nodes`)
+      Assert.isFalse(leafCount < 0)
```

Comment:
This function `pastRoot` is used in other places besides just verification. That's why we don't throw a `VerifyError` and we throw a generic error instead. `VerificationResultReason.INVALID_SPEND` also does not make sense in this function because this function is used for blocks as well, not just spends.

Reply:
I think the solution is not to just use a generic error that never gets caught. If you believe that is not viable to use `VerifyError` then a different (more related?) error type should be used), and that error type should be caught.
And about the use of `INVALID_SPEND`, I couldn't find an error reason from the existing ones that better described the error. Perhaps an additional error reason could be added (i.e., INVALID_TREE_INDEX)?